mobly/controllers/android_device_lib/snippet_client.py

```diff
@@ -32,42 +31,40 @@ class Error(Exception):
 
 
 class SnippetClient(jsonrpc_client_base.JsonRpcClientBase):
-    def __init__(self, package, port, adb_proxy):
+    def __init__(self, package, host_port, adb_proxy):
         """Initialzies a SnippetClient.
   
         Args:
           package: (str) The package name of the apk where the snippets are
             defined.
-          port: (int) The port at which to start the snippet client. Note that
-            the same port will currently be used for both the device and host
-            side of the connection.
+          host_port: (int) The port at which to start the snippet client. Note
+            that the same port will currently be used for both the device and
+            host side of the connection.
             TODO(adorokhine): allocate a distinct free port for both sides of
             the connection; it is not safe in general to reuse the same one.
           adb_proxy: (adb.AdbProxy) The adb proxy to use to start the app.
         """
-        super(SnippetClient, self).__init__(adb_proxy)
+        super(SnippetClient, self).__init__(host_port, adb_proxy)
         self.app_name = package
-        self._package = package
-        self._port = port
-
-    @property
-    def package(self):
-        return self._package
+        # TODO(adorokhine): Don't assume that a free host-side port is free on
+        # the device as well. Both sides should allocate a unique port.
+        self.device_port = host_port
+        self.package = package
```

Comment:
I know the point of nontrivial computation, which was the standard argument, in py2.
In py3, setting a property is not allowed. So it's still better to wrap it in property if it's something we don't want users to be able to accidentally change?

```
>>> class A:
...   def __init__(self):
...     self._var = 40
...   @property
...   def var(self):
...     return self._var
...
>>> a = A()
>>> a.var = 42
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: can't set attribute
```